app/actions/remote/groups.ts

```diff
@@ -1,104 +1,132 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
+import {storeGroups} from '@actions/local/group';
+import {prepareGroups} from '@app/queries/servers/group';
 import {Client} from '@client/rest';
+import DatabaseManager from '@database/manager';
 import NetworkManager from '@managers/network_manager';
 
 import {forceLogoutIfNecessary} from './session';
 
-export const fetchGroupsForChannel = async (serverUrl: string, channelId: string) => {
-    let client: Client;
+export const fetchGroupsForAutocomplete = async (serverUrl: string, query: string, fetchOnly = false) => {
     try {
-        client = NetworkManager.getClient(serverUrl);
-        return client.getAllGroupsAssociatedToChannel(channelId);
-    } catch (error) {
-        forceLogoutIfNecessary(serverUrl, error as ClientErrorProps);
-        return {error};
-    }
-};
+        const client: Client = NetworkManager.getClient(serverUrl);
+        const response = await client.getGroups(query);
 
-export const fetchGroupsForMembership = async (serverUrl: string, userId: string) => {
-    let client: Client;
-    try {
-        client = NetworkManager.getClient(serverUrl);
-        return client.getAllGroupsAssociatedToMembership(userId);
+        // Save locally
+        if (!fetchOnly) {
+            return await storeGroups(serverUrl, response);
+        }
+
+        const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
+        if (!operator) {
+            throw new Error(`${serverUrl} operator not found`);
+        }
```

Comment:
Not to spoil the party but this sort of checks should happen earlier, that way you save computing time and avoid executing requests